src/main/kotlin/no/item/xp/plugin/parser/TextLineParser.kt

```diff
@@ -0,0 +1,56 @@
+package no.item.xp.plugin.parser
+
+import javax.xml.xpath.XPath
+import javax.xml.xpath.XPathConstants
+import javax.xml.xpath.XPathExpression
+import javax.xml.xpath.XPathFactory
+import no.item.xp.plugin.models.GeneratedField
+import no.item.xp.plugin.models.InputType
+import org.w3c.dom.Node
+import org.w3c.dom.NodeList
+
+open class TextLineParser {
+  companion object Factory {
+    @Suppress("unused")
+    fun create(): TextLineParser = TextLineParser()
+  }
+  open fun parseTextLine(node: Node): GeneratedField {
+    var optional = false
+    var comment: String? = null
+    var name: String? = null
+    var type = ""
+    var minimum: Int? = null
+    val nodeList: NodeList = node.childNodes
+    for (i: Int in 0 until nodeList.length) {
```

Comment:
I find this too difficult to read. Replace loops with functions on `Sequence`, and pull outs functions where neccesary.

Reply:
Implemented a foreach-function for the nodelist.